### PR TITLE
Prefer conditional waiting over magic sleep

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [x] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by nrb — DISMISSED** [review]

**Comment by ashish-amarnath** [comment]
Partially addresses #3532

**Review by zubron — CHANGES_REQUESTED** [review]
This is great, thank you @ashish-amarnath! I just have one request to also wait for the namespace deletion at the end of `RunKibishiiTests`.

File: changelogs/unreleased/3527-ashish-amarnath [hunk 1]
```diff
@@ -0,0 +1 @@
+Prefer conditional waiting over magic sleep.
```
[the remark this answers is not in the document]
Comment (ashish-amarnath) [reply changelogs/unreleased/3527-ashish-amarnath:1]: This IMO is a bug fix

[the remark this answers is not in the document]
Comment (carlisia) [reply changelogs/unreleased/3527-ashish-amarnath]: What @zubron said. The CI check is for all feature changes which do need a changelog so we can compile them later for the release notes. Everything else gets labeled nochangelogneeded, document, or website to skip the CI check.

Adding changelogs for tests only adds unnecessary clutter to what will need to be release notes later.

[the remark this answers is not in the document]
Comment (carlisia) [reply changelogs/unreleased/3527-ashish-amarnath]: I could swear I wrote the heuristics for what PR type gets what label, but looked everywhere and didn't find it.

**Comment by ashish-amarnath** [comment]
The change log file has now been removed!

________________________________

**Review by zubron — APPROVED** [review]

**Review by carlisia — APPROVED** [review]
This looks good 👍

**Pull request merged**